sr_utilities_common/scripts/sr_utilities_common/performance_test.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/python
+#
+# Copyright (C) 2018 Shadow Robot Company Ltd - All Rights Reserved.
+# Proprietary and Confidential. Unauthorized copying of the content in this file, via any medium is strictly prohibited.
+
+import rospy
+import argparse
+from sr_utilities.hand_finder import HandFinder
+from diagnostic_msgs.msg import DiagnosticArray
+from sr_robot_msgs.msg import EthercatDebug
+from copy import deepcopy
+
+
+class PerformanceTest(object):
+    def __init__(self):
+        rospy.loginfo("Performance test started")
+        self._hand_finder = HandFinder()
+        self._hand_e = self._hand_finder.hand_e_available()
+        self._hand_h = self._hand_finder.hand_h_available()
+        rospy.loginfo("Hand E available: {}".format(self._hand_e))
+        rospy.loginfo("Hand H available: {}".format(self._hand_h))
+
+        rospy.Subscriber("/diagnostics_agg", DiagnosticArray, self._diagnostics_agg_callback)
+        if self._hand_e:
+            rospy.Subscriber("/rh/debug_etherCAT_data", EthercatDebug, self._ethercat_data_callback)
+
+        self.total_control_loop_overruns_count = 0
+        self.total_recent_control_loop_overruns_count = 0
+        self.total_invalid_packets_total_count = list()
+        self.total_invalid_packets_recent = 0
+        self._invalid_hand_e_packets_total = 0
+
+    def _diagnostics_agg_callback(self, diagnostics_full_msg):
+        diagnostics_control_loop_msg = None
+        diagnostics_ethercat_msg_list = list()
+
+        for msg in diagnostics_full_msg.status:
+            if msg.name == "/Realtime Control Loop/Realtime Control Loop":
+                diagnostics_control_loop_msg = msg
+
+            if msg.name.startswith("/EtherCat/EtherCAT Slaves/H0_"):
+                diagnostics_ethercat_msg_list.append(msg)
+
+        self._set_control_loop_values(diagnostics_control_loop_msg)
+        if self._hand_h:
+            self._set_ethercat_values(diagnostics_ethercat_msg_list)
+
+    def _ethercat_data_callback(self, msg):
+        msg_list = list()
```

Comment:
You are resetting the list every time you receive a msg.
Why have a list if you do that?